src/main/scala/it/pagopa/interop/agreementprocess/service/CatalogManagementService.scala

```diff
@@ -37,13 +37,24 @@ object CatalogManagementService {
     descriptorId: UUID,
     allowedStates: List[EServiceDescriptorState]
   ): Future[Unit] = {
-    val descriptorStatus = eService.descriptors.find(_.id == descriptorId).map(_.state)
+    val validDescriptorState = List(EServiceDescriptorState.SUSPENDED, EServiceDescriptorState.PUBLISHED)
+    val descriptorStatus     =
+      eService.descriptors
+        .filterNot(_.state == EServiceDescriptorState.DRAFT)
+        .sortBy(_.version.toLong)
+        .lastOption
```

Comment:
just FYI, this can be replaced with
```suggestion
        .maxByOption(_.version.toLong)
```